### ISSUE TITLE
Add copy / paste

### ISSUE THREAD

**Issue by futurepaul** [xi-editor/druid#98]
As we work toward the "longer term future" as described in https://github.com/xi-editor/druid/issues/16, a pasteboard (the Mac word) / clipboard (the Windows word) would very nice to have!

One possibility is to rely on the [Clipboard crate](https://crates.io/crates/clipboard). In fact, it seems like it's not totally necessary for this functionality to be inside druid-shell or even druid? To my understanding, [the winit project declared clipboard functionality out of scope](https://github.com/rust-windowing/winit/issues/162) for this reason.

In the longer term, I would expect a clipboard that supports data types beyond text, or a non-global pasteboard, will be important for apps like Runebender. But maybe that's running before we can walk.

**Comment by cmyr** [comment]
I'm going to hijack this issue to track copy/paste more generally; we're going to need to copy and paste paths for runebender, and if we're doing that within the app we should take the time to get it working with the rest of the system. For paths this means converting them to postscript.

Not sure about what the API for this will look like, but at the very least we should try to support text, rich text, images (at least png+jpg), and paths.

**Comment by raphlinus** [comment]
I don't have strong feelings whether support is in druid-shell proper or not. It seems like there's benefit to having it hook the event loop. But yes, a valuable feature, and also valuable to support types beyond plain text (including vector graphics), as they're directly useful for Runebender.

As we've discussed on Zulip, it's unclear whether EPS or PDF are a better type for vector paths. The former is [deprecated](https://developer.apple.com/documentation/appkit/nspasteboard/pasteboardtype) but possibly more widely implemented in other tools that we care about interop with.

**Comment by cmyr** [comment]
I think this will require druid-shell support, because the exact mechanics of the pasteboard (such as how the type of data is communicated) will be dependent on the platform.
